integreat_cms/api/v3/socialmedia_headers.py

```diff
@@ -0,0 +1,185 @@
+    if not region:
+        return root_socialmedia_headers(request, language, path or "")
+
+    if len(slugs) < 3:
+        return region_socialmedia_headers(request, region, path or "")
+
+    return page_socialmedia_headers(request, region, language, "/".join(slugs[2:]))
+
+
+def root_socialmedia_headers(
+    request: HttpRequest, language: Language, path: str
+) -> HttpResponse:
+    """
+    Renders the socialmedia headers for a root page
+
+    :param request: The request that has been sent to the django server
+    :param language: The language of the page
+    :param path: The path to the page
+
+    """
+    title = language.socialmedia_webapp_title
+    url = f"{settings.WEBAPP_URL}/{path}"
+
+    excerpt = (
+        (
+            "hallo aschaffenburg is your digital companion for the town of Aschaffenburg. Here"
+            " you can find local information, advice centres and services."
+        )
+        if settings.BRANDING == "aschaffenburg"
+        else (
+            settings.BRANDING_TITLE + " is your digital companion for Germany. Here"
+            " you can find local information, advice centres and services."
+        )
+    )
+    return render_socialmedia_headers(request, title, excerpt, url)
+
+
+def region_socialmedia_headers(
+    request: HttpRequest, region: Region, path: str
+) -> HttpResponse:
+    """
+    Renders the social media headers for a root region page
+
+    :param request: The request that has been sent to the Django Server
+    :param region: The region the request refers to
+    :param path: The path to the page
+    """
+    title = f"{region.name} | {settings.BRANDING_TITLE}"
+    url = f"{settings.WEBAPP_URL}/{path}"
+    return render_socialmedia_headers(request, title, None, url)
+
+
+def page_socialmedia_headers(
+    request: HttpRequest, region: Region, language: Language, path: str
+) -> HttpResponse:
+    """
+    Renders the social media headers for a single page
+
+    :param request: The request that has been sent to the Django server
+    :param region: The region the page belongs to
+    :param language: Language of the page
+    :param path: Should be a page translation slug
+    :return: HTML meta headers required by social media platforms
+    """
+
+    route, *route_path = path.split("/", maxsplit=1)
+
+    if route == "events" and len(route_path):
+        # TODO: add event json-ld
+        content_translation = EventTranslation.search(
+            region, language.slug, *route_path
+        ).first()
+    elif route == "locations" and len(route_path):
+        content_translation = POITranslation.search(
+            region, language.slug, *route_path
+        ).first()
+    elif route == "news" and len(route_path):
+        content_translation = PushNotificationTranslation.search(
+            region, language.slug, *route_path
+        ).first()
+    else:
+        # TODO: add breadcrumb json-ld if content_translation exists
```

Comment:
Is it okay to leave the todos? I'd prefer doing json-ld in a separate PR

Reply:
Sure :+1: